src/UID2.Client/Site.cs

```diff
@@ -10,7 +10,7 @@ internal class Site
         public Site(int id, IEnumerable<string> domainNames)
         {
             Id = id;
-            _domainNames = new HashSet<string>(domainNames, StringComparer.OrdinalIgnoreCase);
+            _domainNames = new HashSet<string>(domainNames);
```

Comment:
I did not change the variable name and method name inside the site class because the json format `site_data` has the field called `domain_names`